pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -332,13 +334,23 @@ func (p *PriorityQueue) PreEnqueue(pInfo *framework.QueuedPodInfo) bool {
 // Note: we need to associate the failed plugin to `pInfo`, so that the pod can be moved back
 // to activeQ by related cluster event.
 func (p *PriorityQueue) runEnqueuePlugins(ctx context.Context, pInfo *framework.QueuedPodInfo) bool {
+	var s *framework.Status
 	pod := pInfo.Pod
+	startTime := time.Now()
+	defer func() {
+		metrics.FrameworkExtensionPointDuration.WithLabelValues(enqueue, s.Code().String(), pod.Spec.SchedulerName).Observe(metrics.SinceInSeconds(startTime))
+	}()
+
 	for _, pl := range p.enqueuePluginMap[pod.Spec.SchedulerName] {
-		s := pl.PreEnqueue(ctx, pod)
+		s = pl.PreEnqueue(ctx, pod)
 		if s.IsSuccess() {
 			continue
 		}
+		if pInfo.UnschedulablePlugins == nil {
+			pInfo.UnschedulablePlugins = sets.NewString()
```

Comment:
shouldn't this be initialized on pInfo creation?

Reply:
yes, it was for adapting to a ut. Updated the UT now.